test/net/sourceforge/kolmafia/request/AscensionRequestTest.java

```diff
@@ -0,0 +1,160 @@
+package net.sourceforge.kolmafia.request;
+
+import static internal.helpers.Networking.html;
+import static internal.helpers.Player.withNextResponse;
+import static internal.helpers.Player.withProperty;
+import static internal.matchers.Preference.isSetTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import internal.helpers.Cleanups;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AscensionRequestTest {
+  @BeforeEach
+  public void initializeCharPrefs() {
+    KoLCharacter.reset("fakePrefUser");
+    KoLCharacter.reset(true);
+    Preferences.saveSettingsToFile = false;
+    // Fix another test not cleaning up
+    FightRequest.currentRound = 0;
+  }
+
+  @AfterEach
+  public void resetCharAndPrefs() {
+    KoLCharacter.reset("");
+    KoLCharacter.reset(true);
+    KoLCharacter.setUserId(0);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @Test
+  void testAscensionsTodayTracked() {
+    // Simple test to confirm that it resets our last run breakfast
+    String breakfastPref = "lastBreakfast";
+    int breakfastBefore = 5;
+    int breakfastAfter = 0;
+    int breakfastValhalla = -1;
+    String ascensionsPref = "ascensionsToday";
+    int ascensionBefore = 1;
+    int ascensionAfter = 2;
+    var cleanups =
+        new Cleanups(
+            withProperty(breakfastPref, breakfastBefore),
+            withProperty(ascensionsPref, ascensionBefore));
+
+    try (cleanups) {
+      var gashCleanups = withNextResponse(302, Map.of("location", List.of("afterlife.php")), "");
```

Comment:
I want this so bad. I'm currently working on code that does a redirect and was pondering how to test it. I was about to look into modifying our fake client - and you have done it for me!

Want want want.